demoapp/demo_app.py

```diff
@@ -58,224 +58,430 @@ def get_mgl_sections_file() -> pd.DataFrame:
-#         # df.loc[mask, "Tags"] = tag
-#         df.loc[mask, "Rouge-1"] = f"{rouge_scorer['rouge1'].fmeasure:.2f}"
-#         df.loc[mask, "Rouge-2"] = f"{rouge_scorer['rouge2'].fmeasure:.2f}"
-#         df.loc[mask, "Rouge-L"] = f"{rouge_scorer['rougeL'].fmeasure:.2f}"
-#         df.loc[mask, "Cosine Similarity"] = cosine_sim_score
-#         df.loc[mask, "Fact Consistency"] = fact_const_score
+    mgl_documents = [Document(page_content=x) for x in mgl_ref]
+  
+    try:
+        bill_text = [Document(page_content=x) for x in bill_text]
+    except Exception as e:
+        print("No Bill Text")
         
-#     else:
-#         new_bill = pd.DataFrame(columns=["Bill Number", "Bill Title", "Summarized Bill", "Category", "Rouge-1", "Rouge-2", "Rouge-L, Cosine Similarity, Fact Consistency"])
-#         df = pd.concat([df, new_bill], ignore_index=True)
     
-#     df.to_csv(csv_file_path, index=False)
-#     return df
+    if mgl_documents is None or str(mgl_documents).strip().lower()  == "nan" or mgl_documents == "":
+        template = """Can you please explain what the following MA bill means to a regular resident without specialized knowledge? 
+                Please provide a one paragraph summary in 4 sentences. \n Please be simple, direct and concise for the busy reader. Make bullet points if possible.\n
+                Note that the bill refers to specific existing chapters and sections of the Mass General Laws. \n 
+                Summarize the bill that reads as follows {context} and has a title {title}
+                
+                After generating summary, classify the bill according to this list of {categories} to the closest relevant category. Do not output categories outside of the list. \n
+                Then, output top 3 tags in this specific category from the list of tags {tags} that are relevant to this bill. \n
+                Do not output the tags outside from the list. \n
+    
+                Use the corresponding {names} of Mass General Law chapter and sections for constructing your summary, generating categories and tags.\n 
+                Response Format:\nSummary:\n[SUMMARY]\nCategory:\n[CATEGORY]\nTags:\n[TAGS]")
 
-# csv_file_path = "demoapp/new_bills1.csv"
+                """
+        prompt = PromptTemplate.from_template(template)
 
+        chain = create_stuff_documents_chain(llm, prompt)
 
-answer_container = st.container()
-with answer_container:
-    submit_button = st.button(label='Summarize')
-    col1, col2, col3 = st.columns([1.5, 1.5, 1])
+        with get_openai_callback() as cb:
+            response = chain.invoke({"context": bill_text, "title": bill_title, "names": mgl_names, "categories": category_for_bill, "tags": tags_for_bill})
+            st.write(cb.total_tokens, cb.prompt_tokens, cb.completion_tokens, cb.total_cost)
+    else:
+        template =  """Can you please explain what the following MA bill means to a regular resident without specialized knowledge? 
```

Comment:
Can we avoid repeating the template here? I can't tell easily what is different from above. Can we have a single template accomodating both use cases?